pkg/apis/serving/v1beta1/revision_types.go

```diff
@@ -0,0 +1,136 @@
+// references a container image. Revisions are created by updates to a
+// Configuration.
+//
+// See also: https://github.com/knative/serving/blob/master/docs/spec/overview.md#revision
+type Revision struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +optional
+	Spec RevisionSpec `json:"spec,omitempty"`
+
+	// +optional
+	Status RevisionStatus `json:"status,omitempty"`
+}
+
+// Verify that Revision adheres to the appropriate interfaces.
+var (
+	// Check that Revision can be validated, can be defaulted, and has immutable fields.
+	_ apis.Validatable = (*Revision)(nil)
+	_ apis.Defaultable = (*Revision)(nil)
+	_ apis.Immutable   = (*Revision)(nil)
+
+	// Check that we can create OwnerReferences to a Revision.
+	_ kmeta.OwnerRefable = (*Revision)(nil)
+)
+
+// RevisionTemplateSpec describes the data a revision should have when created from a template.
+// Based on: https://github.com/kubernetes/api/blob/e771f807/core/v1/types.go#L3179-L3190
+type RevisionTemplateSpec struct {
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +optional
+	Spec RevisionSpec `json:"spec,omitempty"`
+}
+
+// RevisionContainerConcurrencyType is an integer expressing a number of
+// in-flight (concurrent) requests.
+type RevisionContainerConcurrencyType int64
+
+const (
+	// The maximum configurable container concurrency.
+	RevisionContainerConcurrencyMax RevisionContainerConcurrencyType = 1000
+)
+
+// RevisionSpec holds the desired state of the Revision (from the client).
+type RevisionSpec struct {
+	// RevisionSpec inlines PodSpec and restricts the fields it allows
+	// via webhook.
+	corev1.PodSpec `json:",inline"`
+
+	// ContainerConcurrency specifies the maximum allowed
+	// in-flight (concurrent) requests per container of the Revision.
+	// Defaults to `0` which means unlimited concurrency.
+	// This field replaces ConcurrencyModel. A value of `1`
+	// is equivalent to `Single` and `0` is equivalent to `Multi`.
+	// +optional
+	ContainerConcurrency RevisionContainerConcurrencyType `json:"containerConcurrency,omitempty"`
+
+	// TimeoutSeconds holds the max duration the instance is allowed for responding to a request.
+	// +optional
+	TimeoutSeconds int64 `json:"timeoutSeconds,omitempty"`
+}
+
+const (
+	// RevisionConditionReady is set when the revision is starting to materialize
+	// runtime resources, and becomes true when those resources are ready.
+	RevisionConditionReady = apis.ConditionReady
+)
+
+// RevisionStatus communicates the observed state of the Revision (from the controller).
+type RevisionStatus struct {
+	duckv1beta1.Status `json:",inline"`
+
+	// ServiceName holds the name of a core Kubernetes Service resource that
+	// load balances over the pods backing this Revision. When the Revision
+	// is Active, this service would be an appropriate ingress target for
+	// targeting the revision.
+	// +optional
+	ServiceName string `json:"serviceName,omitempty"`
```

Comment:
I know we currently have this in v1alpha1, but why do we have this as a field instead of a label or annotation? It seems like an implementation detail we are exposing in our API and it is not clear to me why a user needs it here.

Reply:
Do you want to propose an alternate name?